polaris/polaris/deposit/views.py

```diff
@@ -141,50 +139,69 @@ def confirm_transaction(account: str, request: Request) -> Response:
     """
     # Validate query parameters: account, asset_code, transaction_id.
     account = request.GET.get("account")
     asset_code = request.GET.get("asset_code")
     asset = Asset.objects.filter(code=asset_code).first()
     transaction_id = request.GET.get("transaction_id")
     if not account:
-        return render_error_response("no 'account' provided", content_type="text/html")
+        err_msg = "no 'account' provided"
+        return render_error_response(err_msg, content_type="text/html")
     elif not (asset_code and asset):
-        return render_error_response("invalid 'asset_code'", content_type="text/html")
+        err_msg = "invalid 'asset_code'"
+        return render_error_response(err_msg, content_type="text/html")
     elif not transaction_id:
-        return render_error_response("no 'transaction_id' provided", content_type="text/html")
+        err_msg = "no 'transaction_id' provided"
+        return render_error_response(err_msg, content_type="text/html")
 
+    # Ensure the transaction exists
     try:
         transaction = Transaction.objects.get(
-            id=transaction_id, asset=asset
+            id=transaction_id,
+            asset=asset
         )
-    except Transaction.DoesNotExist:
+    except Transaction.objects.DoesNotExist:
         return render_error_response(
             "Transaction with ID and asset_code not found",
             content_type="text/html",
             status_code=status.HTTP_404_NOT_FOUND
         )
 
     if request.method == "GET":
-        form = DepositForm()
+        form = rdi.form_for_transaction(transaction)()
         return Response({"form": form}, template_name="deposit/form.html")
 
-    form = DepositForm(request.POST)
-    form.asset = asset
-    # If the form is valid, we create a transaction pending external action
-    # and render the success page.
+    # request.method == "POST"
+    form = rdi.form_for_transaction(transaction)(request.POST)
+    is_transaction_form = issubclass(form.__class__, TransactionForm)
+    if is_transaction_form:
+        form.asset = asset
+
     if form.is_valid():
-        transaction.amount_in = form.cleaned_data["amount"]
-        transaction.amount_fee = calc_fee(
-            asset, settings.OPERATION_DEPOSIT, transaction.amount_in
-        )
-        transaction.status = Transaction.STATUS.pending_user_transfer_start
-        transaction.save()
+        if is_transaction_form:
+            transaction.amount_in = form.cleaned_data["amount"]
+            transaction.amount_fee = calc_fee(
+                asset, settings.OPERATION_DEPOSIT, transaction.amount_in
+            )
+            transaction.save()
+
+        # Perform any defined post-validation logic defined by Polaris users
+        rdi.after_form_validation(form, transaction)
+        # Check to see if there is another form to render
+        form_class = rdi.form_for_transaction(transaction)
+
+        if form_class:
+            return Response(
```

Comment:
I get the feeling the GET and POST methods should be different endpoints, both for code clarity, but also in case the page needs to reload.  
For example `POST /webapp/submit_form` will process the request then redirect to `GET /webapp` to show the next available form.  Otherwise, if the page needs to reload for any reason the browser/webview will want to rePOST the previous form.

I think this is not a big deal for now but we may want to change this in the future

Reply:
Thats fair, I'll make a separate PR for it. #34